### ISSUE TITLE
Wheel support for linux aarch64

### ISSUE THREAD

**Issue by odidev** [giampaolo/psutil#1782]
**Summary**
Installing psutil on aarch64 via pip using command "pip3 install psutil" tries to build wheel from source code

**Problem description**
psutil don't have wheel for aarch64 on PyPI repository. So, while installing psutil via pip on aarch64, pip builds wheel for same resulting in it takes more time to install psutil. Making wheel available for aarch64 will benefit aarch64 users by minimizing psutil installation time.

**Expected Output**
Pip should be able to download psutil wheel from PyPI repository rather than building it from source code.

@psutil-taem, please let me know if I can help you building wheel/uploading to PyPI repository. I am curious to make psutil wheel available for aarch64. It will be a great opportunity for me to work with you.

**Comment by odidev** [comment]
Gentle Reminder!!!

**Comment by SoraDevin** [comment]
gentle but firm reminder

**Comment by giampaolo** [comment]
Looking back at this: in general I'm not planning on supporting / distributing wheels for `aarch64` at the moment. The current build matrix run tests + generates wheels on the following [platform / arch / python-version] combinations:

```
windows (total = 10)                                      size    arch   pyver
psutil-5.8.1-cp27-none-win32.whl                        230.9K      32    py27
psutil-5.8.1-cp27-none-win_amd64.whl                    234.2K      64    py27
psutil-5.8.1-cp36-cp36m-win32.whl                       234.7K      32    py36
psutil-5.8.1-cp36-cp36m-win_amd64.whl                   239.1K      64    py36
psutil-5.8.1-cp37-cp37m-win32.whl                       234.7K      32    py37
psutil-5.8.1-cp37-cp37m-win_amd64.whl                   239.1K      64    py37
psutil-5.8.1-cp38-cp38-win32.whl                        235.3K      32    py38
psutil-5.8.1-cp38-cp38-win_amd64.whl                    239.7K      64    py38
psutil-5.8.1-cp39-cp39-win32.whl                        235.8K      32    py39
psutil-5.8.1-cp39-cp39-win_amd64.whl                    240.4K      64    py39

macos (total = 5)                                         size    arch   pyver
psutil-5.8.1-cp27-cp27m-macosx_10_9_x86_64.whl          230.3K      64    py27
psutil-5.8.1-cp36-cp36m-macosx_10_9_x86_64.whl          230.6K      64    py36
psutil-5.8.1-cp37-cp37m-macosx_10_9_x86_64.whl          230.6K      64    py37
psutil-5.8.1-cp38-cp38-macosx_10_9_x86_64.whl           230.9K      64    py38
psutil-5.8.1-cp39-cp39-macosx_10_9_x86_64.whl           230.8K      64    py39

linux (total = 12)                                        size    arch   pyver
psutil-5.8.1-cp27-cp27m-manylinux2010_i686.whl          277.7K      32    py27
psutil-5.8.1-cp27-cp27m-manylinux2010_x86_64.whl        280.9K      64    py27
psutil-5.8.1-cp27-cp27mu-manylinux2010_i686.whl         277.7K      32    py27
psutil-5.8.1-cp27-cp27mu-manylinux2010_x86_64.whl       280.9K      64    py27
psutil-5.8.1-cp36-cp36m-manylinux2010_i686.whl          282.4K      32    py36
psutil-5.8.1-cp36-cp36m-manylinux2010_x86_64.whl        284.9K      64    py36
psutil-5.8.1-cp37-cp37m-manylinux2010_i686.whl          283.4K      32    py37
psutil-5.8.1-cp37-cp37m-manylinux2010_x86_64.whl        289.3K      64    py37
psutil-5.8.1-cp38-cp38-manylinux2010_i686.whl           286.8K      32    py38
psutil-5.8.1-cp38-cp38-manylinux2010_x86_64.whl         289.0K      64    py38
psutil-5.8.1-cp39-cp39-manylinux2010_i686.whl           284.1K      32    py39
psutil-5.8.1-cp39-cp39-manylinux2010_x86_64.whl         286.5K      64    py39

freebsd + py2.7
freebsd + py3.x

total = 29 unit test runs + 27 wheels
```

The above is triggered on each commit, and it's quite costly both in terms of build time and maintenance, especially when we get close to a release (I already removed the PYPY matrix because of this). Adding yet another platform/arch (also sort of exotic in this case) to the matrix, make sure it works, fix tests, guarantee that the distributed wheels work etc. is extra work I'm unwilling to take to be honest.

**Comment by sebasv** [comment]
I'd like to point out that ARM is here to stay, Apple is releasing more and more devices with ARM processors and I'm struggling to find a workaround

**Comment by glenn-jocher** [comment]
+1

**Comment by gilbahat** [comment]
With graviton 1/2/3 and Ampere altra becoming standard on mainstream clouds and with aarch64 being the de-facto android architecture of choice. I vote +1.

Maybe it's time to put i686 to rest in lieu?

**Comment by jamel** [comment]
One more vote to add support of aarch64. It becomes more and more popular with Apple's M1/2 and in major cloud providers including AWS, GCP and Azure.

**Comment by SoraDevin** [comment]
AWS Panorama is built on it too.

**Comment by jasonrichdarmawan** [comment]
NVIDIA Jetson Nano B01 uses aarch64. We planned to use anomalib. psutil is getting in the way.

**Comment by timorkal** [comment]
why is this closed?

**Comment by jgehrcke** [comment]
In https://pypi.org/project/psutil/5.9.4/#files I see `psutil-5.9.4-cp38-abi3-macosx_11_0_arm64.whl`. Why is that not sufficient? I also think it's not enough, because elsewhere we've been running into a situation where a build step on MacOS/ARM64 did not download the arm64 wheel:

From [that](https://github.com/conbench/conbench/issues/709#issuecomment-1423903187) build log
```
#0 5.866 Collecting numpy
#0 5.887   Downloading numpy-1.24.2-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (14.0 MB)
...
#0 8.377 Collecting psutil
#0 8.405   Downloading psutil-5.9.4.tar.gz (485 kB)
```

In that environment pip was happy with selecting the `aarch64` wheel for numpy, but the arm64 wheel for `psutil-5.9.4` wasn't good enough and it downloaded the source distribution instead.

Everywhere I read that

> "AArch64" and "ARM64" are different names for the same thing

but maybe `pip` is more specific. From the build log excerpt above it's kind of apparent that we want a `*_aarch64.manylinux2014_aarch64.whl` wheel for psutil :).

**Comment by chlndh** [comment]
And building it manually results in an error: `gcc: error: unrecognized debug output level ‘mlt’`

**Comment by Manouchehri** [comment]
Not sure why this was closed as complete, it's not a thing yet as far as I can tell.

Would be nice to be able to deploy arm64 Lambdas that use psutil.

**Comment by SoraDevin** [comment]
@giampaolo reopen this please

**Comment by michielswaanen** [comment]
Same issue here, Dockerfile with a specific arm64/v8 arch not build because of psutil.

@giampaolo 

Dockerfile build error
```Dockerfile
109.8 Building wheels for collected packages: psutil
109.8   Building wheel for psutil (pyproject.toml): started
112.2   Building wheel for psutil (pyproject.toml): finished with status 'error'
112.3   error: subprocess-exited-with-error
112.3
112.3   × Building wheel for psutil (pyproject.toml) did not run successfully.
112.3   │ exit code: 1
112.3   ╰─> [43 lines of output]
112.3       running bdist_wheel
112.3       running build
112.3       running build_py
112.3       creating build
112.3       creating build/lib.linux-aarch64-cpython-310
112.3       creating build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_pslinux.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_common.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_pssunos.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_psosx.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_compat.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_pswindows.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_psaix.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_psbsd.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       copying psutil/_psposix.py -> build/lib.linux-aarch64-cpython-310/psutil
112.3       creating build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_process.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_osx.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_linux.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_bsd.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_memleaks.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_testutils.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_system.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_misc.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_contracts.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_windows.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/__main__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_posix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_sunos.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_unicode.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/runner.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_connections.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       copying psutil/tests/test_aix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
112.3       running build_ext
112.3       building 'psutil._psutil_linux' extension
112.3       creating build/temp.linux-aarch64-cpython-310
112.3       creating build/temp.linux-aarch64-cpython-310/psutil
112.3       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=596 -DPy_LIMITED_API=0x03060000 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -DPSUTIL_LINUX=1 -I/usr/local/include/python3.10 -c psutil/_psutil_common.c -o build/temp.linux-aarch64-cpython-310/psutil/_psutil_common.o
112.3       psutil could not be installed from sources because gcc is not installed. Try running:
112.3         sudo apt-get install gcc python3-dev
112.3       error: command 'gcc' failed: No such file or directory
112.3       [end of output]
112.3
112.3   note: This error originates from a subprocess, and is likely not a problem with pip.
112.3   ERROR: Failed building wheel for psutil
112.3 Failed to build psutil
112.3 ERROR: Could not build wheels for psutil, which is required to install pyproject.toml-based projects
```

**Comment by mattriordan1uphealth** [comment]
For those still looking for a solution.  This worked for me on a gravaton based EC2 instance:

sudo yum install gcc python3-devel
pip install --no-binary :all: psutil

https://github.com/giampaolo/psutil/blob/master/INSTALL.rst

**Comment by giampaolo** [comment]
aarch64 wheels are now published on PYPI as of psutil 6.0.0.
